include/boost/numeric/ublas/tensor/dynamic_strides.hpp

```diff
@@ -46,7 +44,7 @@ class basic_strides
                                  "Static error in boost::numeric::ublas::basic_strides: type must be of type integer.");
     static_assert(!std::numeric_limits<typename base_type::value_type>::is_signed,
                                 "Static error in boost::numeric::ublas::basic_strides: type must be of type unsigned integer.");
-    static_assert(std::is_same<__layout,first_order>::value || std::is_same<__layout,last_order>::value,
+    static_assert(std::is_same<__layout,layout::first_order>::value || std::is_same<__layout,layout::last_order>::value,
```

Comment:
I would not use `__layout`. Starting with two underscores is reserved for Implementation. Any problem with `_layout` (Single underscore?)

Reply:
I know but I did not change it because it present already. Even field members should start or end with any kind of underscore for same reason and to follow the convention I used it in the new structs. If @bassoy wants I can change that in all the places. Thanks for pointing it out I was meaning to ask @bassoy about it.

Reply:
Resolved

Reply:
My problem with it is if we try to follow a compiler way of declaring functions, structs, or variables it will in the future collide. So we have to stay away from this kind of practice as they are reserved for the compilers, not for others.

Reply:
Yes, you are right because they could be used for the C++ implementation.